src/problems/MetalAdvectionProblem/test_sne.cpp

```diff
@@ -0,0 +1,540 @@
+{
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = geom[lev].ProbLoArray();
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_hi = geom[lev].ProbHiArray();
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const &dx = geom[lev].CellSizeArray();
+
+	AddSupernova(state_new_cc_[lev], prob_lo, prob_hi, dx, userData_, lev);
+}
+
+template <>
+AMREX_GPU_DEVICE AMREX_FORCE_INLINE auto HydroSystem<NewProblem>::GetGradFixedPotential(amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> posvec)
+    -> amrex::GpuArray<amrex::Real, AMREX_SPACEDIM>
+{
+
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> grad_potential;
+	grad_potential[0] = 0.0;
+	grad_potential[1] = 0.0;
+
+	double z = posvec[2];
+
+	// Interpolate to find the accurate g-value from array-- because linterp doesn't work on Setonix
+	size_t i = 0;
+	double x_interp = std::abs(z);
+	while (i < z_data.size() - 1 && x_interp > z_data[i + 1]) {
+		i++;
+	}
+
+	// Perform linear interpolation
+	const Real x1 = z_data[i];
+	const Real x2 = z_data[i + 1];
+	const Real y1 = logg_data[i];
+	const Real y2 = logg_data[i + 1];
+
+	amrex::Real ginterp = (y1 + (y2 - y1) * (x_interp - x1) / (x2 - x1));
+
+	grad_potential[2] = 2. * M_PI * Const_G * rho_dm * std::pow(R0_Gal, 2) * (2. * z / std::pow(R0_Gal, 2)) / (1. + std::pow(z, 2) / std::pow(R0_Gal, 2));
+	grad_potential[2] += 2. * M_PI * Const_G * Sigma_star * (z / z_star) * (std::pow(1. + z * z / (z_star * z_star), -0.5));
+	grad_potential[2] += (z / std::abs(z)) * std::pow(10., ginterp);
+
+	return grad_potential;
+}
+
+// Add Strang Split Source Term for External Fixed Potential Here
+template <> void QuokkaSimulation<NewProblem>::addStrangSplitSources(amrex::MultiFab &mf, int lev, amrex::Real time, amrex::Real dt_lev)
+{
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = geom[lev].ProbLoArray();
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const &dx = geom[lev].CellSizeArray();
+	const Real dt = dt_lev;
+
+	for (amrex::MFIter iter(mf); iter.isValid(); ++iter) {
+		const amrex::Box &indexRange = iter.validbox();
+		auto const &state = mf.array(iter);
+
+		amrex::ParallelFor(indexRange, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+			amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> posvec, GradPhi;
+			double x1mom_new, x2mom_new, x3mom_new;
+
+			const Real rho = state(i, j, k, HydroSystem<NewProblem>::density_index);
+			const Real x1mom = state(i, j, k, HydroSystem<NewProblem>::x1Momentum_index);
+			const Real x2mom = state(i, j, k, HydroSystem<NewProblem>::x2Momentum_index);
+			const Real x3mom = state(i, j, k, HydroSystem<NewProblem>::x3Momentum_index);
+			const Real Egas = state(i, j, k, HydroSystem<NewProblem>::energy_index);
+
+			Real Eint = RadSystem<NewProblem>::ComputeEintFromEgas(rho, x1mom, x2mom, x3mom, Egas);
+
+			posvec[0] = prob_lo[0] + (i + 0.5) * dx[0];
+
+#if (AMREX_SPACEDIM >= 2)
+			posvec[1] = prob_lo[1] + (j + 0.5) * dx[1];
+#endif
+
+#if (AMREX_SPACEDIM >= 3)
```

Comment:
Since the CMakeLists.txt file only lets the user compile this problem for 3D anyway, these ifdefs are probably not necessary.